joern-cli/frontends/rubysrc2cpg/src/main/scala/io/joern/rubysrc2cpg/utils/packageTable.scala

```diff
@@ -0,0 +1,42 @@
+package io.joern.rubysrc2cpg.utils
+
+import scala.collection.mutable
+
+case class MethodTableModel(moduleName: String, methodName: String, parentClassPath: String) {
+  override def equals(other: Any): Boolean = {
+    other match {
+      case that: MethodTableModel => this.methodName == that.methodName
+      case _                      => false
+    }
+  }
+
+  override def hashCode: Int = {
+    methodName.hashCode
+  }
+}
+
+case class PackageContext(moduleName: String, packageTable: PackageTable)
+
+class PackageTable() {
+
+  private val methodTableSet = mutable.HashSet[MethodTableModel]()
```

Comment:
This DS should be a Map as we are doing `get`, `contains` on `methodName`. So the key to the map will be `methodName`

Reply:
Yes changed to HashMap data structure to store method info.